### ISSUE TITLE
[BUG] Subject Area findRESTCall problem with сyrillic letters

### ISSUE THREAD

**Issue by rychagova** [odpi/egeria#5710]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Method createFindQuery() encodes сyrillic letters in searchCriteria parameter   to _  **%D**1%82%D0%BE..._
and after that it all goes in String.format(...)
 ` String findUrlTemplate = urnTemplate + createFindQuery(methodName, findRequest, exactValue, ignoreCase).toString();
             String expandedURL = String.format(serverPlatformURLRoot + findUrlTemplate, serverName, userId);
             `
It makes   findUrlTemplate =      /servers/%s/open-metadata/access-services/subject-area/users/%s/categories?sequencingOrder=ANY&pageSize=500&startingFrom=0&**searchCriteria=%D1%82%D0%BE%D0%BF**&exactValue=false&ignoreCase=true

             

### Expected Behavior

searchCriteria in a findUrlTemplate remains in cyrillic 
..0&searchCriteria=топ&..

### Environment

```markdown
- Egeria:
- OS:
- Java:
- Browser (for UI issues):
- Additional connectors and integration:
```

**Comment by davidradl** [comment]
I think that there are some options 
1) you ensure that the default Locale of the server is sirillic, the String.format method should then work as it picks up the default Locale
2) The alternative is that we need to do is supply a Locale as the first parameter of the String.format() method. We could do this as a subject area omas configuration option - default Locale that data is going to be treated with.
3)We could supply the  locales are localo

**Comment by davidradl** [comment]
Hi there,
Before I look at changing any code, I notice https://stackoverflow.com/questions/26814971/urlencoder-encode-doesnt-work-with-cyrillic - whcih turned out not to be a bug. I am wondering whether %D1%82%D0%BE%D0%BF is correct. We issue useEncoder.encode with utf-8 - I assume that the receiving end would decode this in the Spring layers so the search would be sensible. I have coded a test program with "    System.err.println(URLDecoder.decode( "%D1%82%D0%BE%D0%BF", "utf-8"));"  this gives me топ which appears to be the Russian word for top. I will do a bit more testing.

**Comment by davidradl** [comment]
I have tested this with the UI, and breakpointing. I search for топ this comes through as топ . I do not think that the format is is the issue - in my case using the view server the call fails in the view server . I will investigate further to see why this is.

**Comment by rychagova** [comment]
![image](https://user-images.githubusercontent.com/70315152/134898420-dff82f03-465c-48b3-bed4-e2efbb59e5b2.png)
Sorry, I'm a terrible explanator. 
Ive meant that this expandedURL throws an exception because of encoded searchCriteria (with %D1....) is formatted in  String.format()
And I don't know how to avoid this encoding.  so that I could get expandedURL .

**Comment by rychagova** [comment]
Unfortunately, I couldn't check it either. But it should be working. 
It must be added in the next release? 
Thanks in advance.

**Comment by davidradl** [comment]
Hi there, I have merged the fix - it will be in release 3.2 which is being created currently.
